### PR TITLE
Bg_Gnd_Iceblock and Bg_Ice_Objects (sliding ice blocks)

### PR DESCRIPTION
**Pull request by petrie911**
Figured these actors share enough code it was worth doing together. Both have hard-coded logic for the block positions rather than interacting with the collision. Kind of weird.

### CODE REVIEW

File: src/overlays/actors/ovl_Bg_Gnd_Iceblock/z_bg_gnd_iceblock.c [hunk 1]
```diff
@@ -10,21 +10,19 @@
 
 #define THIS ((BgGndIceblock*)thisx)
 
-void BgGndIceblock_Init(Actor* thisx, GlobalContext* globalCtx);
-void BgGndIceblock_Destroy(Actor* thisx, GlobalContext* globalCtx);
-void BgGndIceblock_Update(Actor* thisx, GlobalContext* globalCtx);
-void BgGndIceblock_Draw(Actor* thisx, GlobalContext* globalCtx);
+typedef enum { GNDICE_IDLE, GNDICE_FALL, GNDICE_HOLE } BgGndIceblockAction;
```
Comment (fig02) [inline src/overlays/actors/ovl_Bg_Gnd_Iceblock/z_bg_gnd_iceblock.c:14]: offset comments

File: src/overlays/actors/ovl_Bg_Ice_Objects/z_bg_ice_objects.c [hunk 2]
```diff
@@ -30,23 +38,209 @@ const ActorInit Bg_Ice_Objects_InitVars = {
...
+        -580, -660, -780, -820, -860, -900, -1087,
+    };
+    static s16 sZStops[7][2] = {
+        { -580, -1087 }, { -780, -1260 }, { -340, -820 }, { -260, -1260 },
+        { -340, -860 },  { -660, -1260 }, { -340, -740 },
+    };
+    static s16 sXStops[7][2] = {
+        { -860, -1580 },  { -1240, -1780 }, { -860, -1680 }, { -860, -1680 },
+        { -1387, -1680 }, { -860, -1200 },  { -860, -1800 },
+    };
```
Comment (fig02) [inline src/overlays/actors/ovl_Bg_Ice_Objects/z_bg_ice_objects.c:87]: kinda feeling like these are too big to be in function, unless it was needed to match. same for the other actor if that one has the same thing

Comment (petrie911) [reply src/overlays/actors/ovl_Bg_Ice_Objects/z_bg_ice_objects.c]: Fair. I'll just put them above that function.

**Review by fig02 — APPROVED** [review]

File: src/overlays/actors/ovl_Bg_Gnd_Iceblock/z_bg_gnd_iceblock.c [hunk 3]
```diff
@@ -10,21 +10,23 @@
...
 
-void func_8087A0C8(BgGndIceblock* this, GlobalContext* globalCtx);
-void func_8087A184(BgGndIceblock* this, GlobalContext* globalCtx);
-void func_8087A248(BgGndIceblock* this, GlobalContext* globalCtx);
-void func_8087A300(BgGndIceblock* this, GlobalContext* globalCtx);
-void func_8087A39C(BgGndIceblock* this, GlobalContext* globalCtx);
+void BgGndIceblock_Init(BgGndIceblock* this, GlobalContext* globalCtx);
+void BgGndIceblock_Destroy(BgGndIceblock* this, GlobalContext* globalCtx);
+void BgGndIceblock_Update(BgGndIceblock* this, GlobalContext* globalCtx);
+void BgGndIceblock_Draw(BgGndIceblock* this, GlobalContext* globalCtx);
```
Comment (Roman971) [inline src/overlays/actors/ovl_Bg_Gnd_Iceblock/z_bg_gnd_iceblock.c:22]: I think these functions should just take `Actor* thisx`?

Comment (petrie911) [reply src/overlays/actors/ovl_Bg_Gnd_Iceblock/z_bg_gnd_iceblock.c:22]: yes they should.

**Review by Roman971 — APPROVED** [review]

**Pull request merged**